net/interfaces/interfaces.go

```diff
@@ -117,17 +120,31 @@ func LocalAddresses() (regular, loopback []netaddr.IP, err error) {
 				if tsaddr.IsTailscaleIP(ip) {
 					continue
 				}
-				if ip.IsLinkLocalUnicast() {
-					continue
-				}
 				if ip.IsLoopback() || ifcIsLoopback {
 					loopback = append(loopback, ip)
+				} else if ip.IsLinkLocalUnicast() || tsaddr.IsULA(ip) {
+					if ip.Is4() {
+						linklocal4 = append(linklocal4, ip)
+					} else {
+						linklocal6 = append(linklocal6, ip)
+					}
 				} else {
-					regular = append(regular, ip)
+					if ip.Is4() {
+						regular4 = append(regular4, ip)
+					} else {
+						regular6 = append(regular6, ip)
+					}
 				}
 			}
 		}
 	}
+	if len(regular4) == 0 {
+		regular4 = linklocal4
+	}
+	if len(regular6) == 0 {
+		regular6 = linklocal6
+	}
+	regular = append(regular4, regular6...)
```

Comment:
This doesn't seem to match the docs: it does the selection of link local per address family, not overall.

So in the very common case we'll end up with real IPv4 + link local IPv6 for all the people who don't have working IPv6 (like 60+%) but their OS supports IPv6 so they have an IPv6 link local address. Then this would start returning v6 link local where it didn't before, which means it'd get reported up to control?

Reply:
Yes, that is true.
- We know of cases where the IPv4 link-local 169.254.x.x is used for connectivity, in AWS Lambda.
- We know of cases where IPv6 ULA addresses are used for connectivity, in Cloud Run.

We don't know of cases where IPv6 fe80:: are actually used for connectivity. We could treat those differently.

Reply:
Made it return IPv4 link-local and IPv6 ULA, did not return IPv6 link-local. Updated doc accordingly.